x-pack/filebeat/input/awss3/s3_objects.go

```diff
@@ -385,9 +414,15 @@ func (p *s3ObjectProcessor) readFile(r io.Reader) error {
 	return nil
 }
 
-func (p *s3ObjectProcessor) publish(ack *awscommon.EventACKTracker, event *beat.Event) {
-	ack.Add()
-	event.Private = ack
+func (p *s3ObjectProcessor) publish(event *beat.Event) {
+	if p.acker != nil {
+		event.Private = p.acker
```

Comment:
also need `p.acker.Add()` first?

Reply:
`p.acker` is `awss3.EventACKTracker`: it has no `Add()`
`p.ackerForPollin` is `awscommon.EventACKTracker`: it has `Add()`